pkg/registryclient/client.go

```diff
@@ -171,15 +174,37 @@ func WithTracing() Option {
 	}
 }
 
-// BuildRemoteOption builds remote.Option based on client.
-func (c *client) BuildRemoteOption(ctx context.Context) remote.Option {
+// BuildCosignRemoteOption builds remote.Option for cosign client.
+func (c *client) BuildCosignRemoteOption(ctx context.Context) remote.Option {
```

Comment:
You can use `BuildGCRRemoteOption` function to get the remote opts used here, then return the remote.Option and error